pkg/arvo/sys/arvo.hoon

```diff
@@ -21,65 +21,156 @@
-++  hobo  |*  a/mold                                    ::  kiss wrapper
-          $?  $%  {$soft p/*}                           ::
-              ==                                        ::
-              a                                         ::
-          ==                                            ::
-++  mark  @tas                                          ::  content type
-++  mash  |=(* (mass +<))                               ::  producing mass
-++  mass  $~  [%$ [%& ~]]                               ::  memory usage
-          (pair cord (each noun (list mash)))           ::
-++  mill  (each vase milt)                              ::  vase+metavase
-++  milt  {p/* q/*}                                     ::  metavase
-++  monk  (each ship {p/@tas q/@ta})                    ::  general identity
-++  muse  {p/@tas q/duct r/arvo s/@ud}                  ::  sourced move
-++  move  {p/duct q/arvo}                               ::  arvo move
-++  ovum  {p/wire q/curd}                               ::  typeless ovum
-++  pane  (list {p/@tas q/vase})                        ::  kernel modules
-++  pone  (list {p/@tas q/vise})                        ::  kernel modules old
+::
++|  %global
+::
+::  $arch: fundamental node
+::  $beak: global context
+::  $beam: global name
+::  $bone: opaque duct handle
+::  $case: global version
+::  $cage: global metadata
+::  +cask: global data builder
+::  $desk: local workspace
+::  $dock: message target
+::  $mark: symbolic content type
+::  $ship: network identity
+::  $sink: subscription
+::
++$  arch  [fil=(unit @uvI) dir=(map @ta ~)]
++$  beam  [beak s=path]
++$  beak  (trel ship desk case)
++$  bone  @ud
++$  case
+  $%  ::  %da:  date
+      ::  %tas: label
+      ::  %ud:  sequence
+      ::
+      [%da p=@da]
+      [%tas p=@tas]
+      [%ud p=@ud]
+  ==
++$  cage  (cask vase)
+++  cask  |$  [a]  (pair mark a)
++$  desk  @tas
++$  dock  (pair @p term)
++$  mark  @tas
++$  ship  @p
+++  sink  (trel bone ship path)
+::
++|  %meta
+::
+::  +hypo: type-associated builder
+::  $meta: meta-vase
+::  $maze: vase, or meta-vase
+::
+++  hypo
+  |$  [a]
+  (pair type a)
++$  meta  (pair)
++$  maze  (each vase meta)
+::
++|  %interface
+::
+::  $ball: dynamic kernel action
+::  $curd: tagged, untyped event
+::  $duct: causal history
+::  +hobo: %soft task builder
+::  $goof: crash label and trace XX fail/ruin/crud/flaw/lack/miss
+::  $monk: general identity
+::  $move: cause and action
+::  $ovum: card with cause
+::  $scry-sample: vane +scry argument
+::  $vane-sample: vane wrapper-gate aargument
+::  $sley: namespace function
+::  $slyd: super advanced
+::  $slyt: old namespace
+::  +wind: kernel action builder
+::  $wire: event pretext
+::  +wite: kernel action/error builder
+::
++$  ball  (wite [vane=term task=maze] maze)
++$  curd  (cask)
++$  duct  (list wire)
+++  hobo
+  |$  [a]
+  $?  $%  [%soft p=*]
+      ==
+      a
+  ==
++$  goof  [mote=term =tang]
++$  monk  (each ship (pair @tas @ta))
```

Comment:
Maybe it's out of scope, but while we're messing with Arvo, it would be nice to remove this (incomplete) definition of a person's name from the kernel.  I don't know how hard it would be to switch scrys to use `(unit (set ship))` instead of `(unit (set monk))`, but it might be pretty quick.

Reply:
It's definitely bad, but I'd prefer to handle the whole scry-sample at once. FWIW, the neo scry-sample is just `[(unit (set ship)) case=term =beam]`.